### DISCUSSION TITLE
Swell.js doesnt return attributes with custom fields

### DISCUSSION THREAD

**Discussion by BramGerrits** [swellstores/community#179] [Help] [answered]
Hi,

For a customer project I'm using the nuxt-swell package (version "0.2.0"). I'm trying to group attributes together by adding a content field called  `attribute_group` to the Attributes model. 

But when I perform `await swell.products.get('<PRODUCT_SLUG>');`, the attribute objects don't come with the `attribute_group` field. The response i'm getting looks like this:

```javascript
// await swell.products.get('<PRODUCT_SLUG>');
{
  attributes: {
    length: {
      filterable: true,
      id: "length",
      name: "Lengte (mm)",
      type: "number",
      value: 600,
      visible: true,
      // no `attribute_group`
    },
    ...
  },
  ...
}

``` 

The `attribute_group` field does appear to exist when i look at the json of the Attributes model in the developer console.

```typescript
{
    "version": "1.0.15",
    "name": "attributes",
    "api": "com",
    "label": "Attribute",
    "plural": "Attributes",
    "fields": {
        "content": {
            "type": "object",
            "fields": {
                "attribute_group": {
                    "type": "string",
                    "public": true,
                    "content_id": "custom.admin.attribute-fields"
                }
            }
        },
        "attributes": {
            "type": "object",
            "fields": {
                "attribute_group": {
                    "type": "string",
                    "content_id": "custom.admin.attribute-fields"
                }
            }
        },
        "attribute_group": {
            "type": "object",
            "fields": {
                "attribute_group": {
                    "type": "string",
                    "content_id": "custom.admin.attribute-fields"
                }
            }
        },
        ...
    },
    ...
}
``` 

Is there a way to retrieve custom content fields with the swell.js package? If no, is there another way to group attributes together?

**Comment by mustafaswell** [comment]
Hello @BramGerrits ,

Thank you for reporting. This seems to be a gap here that you can not expand and get custom content fields using Swell JS.

**Reply by BramGerrits** [reply]
Hi @mustafaswell ,

Thank you for your response 😄 

Unfortunately, this means that the solution for grouping attributes provided in [this discussion](https://github.com/orgs/swellstores/discussions/105#discussion-4784450) doesn't work. Is there another way for grouping attributes together? Also, does Swell have plans for returning the custom fields of attributes with SwellJS in the near future? 

Thanks in advance

**Reply by mustafaswell** [reply]
Good Morning @BramGerrits ,

I've played around with Swell JS and seems like we are not able to drill into the attribute object that is returned when getting a Product. The object will always return the fields: `filterable`, `id`, `name`, `type`, `value`, `visible`. So adding a custom field to the Attribute Object such as `attribute_group` though ideal, will not surface when using `products.get()`.

_Possible Workaround_

The solution described below adds an additional API Call 

1. Create Attribute Group Object which will have a Collection of Linked Attributes.
  - Attribute Group: Sizes
    - Length
    - Width
    - Height
    - Depth
 1. Make another API Call to retrieve the Attribute Groups and determine which attributes belong to which Attribute Group.
  - Since you'll be getting a Custom Model consider using `swell.content.get(<attribute_name>, <id>)`. For example if the Custom Model is "Attribute Group", then it may look like this: `swell.content.get('attribute-group', 'asd324df')`.

**Reply by BramGerrits** [reply]
This sounds like a good solution. However I see that the content editor (`swell.store/admin/content`) is not available yet. Can you tell me when this page will be available?

<img width="1920" alt="Schermafbeelding 2023-03-21 om 16 42 12" src="https://user-images.githubusercontent.com/43609156/226661582-93543cb3-8eed-4a87-8d7f-49e4d42d5b57.png">

**Reply by logeshswell** [reply, marked as answer]
@BramGerrits Another possible workaround would be to fetch and cache all attributes in your memory at once which would contain `content` fields as well if you enabled content permissions for `attributes` model as described below:
https://developers.swell.is/guides/opening-content-permissions

```js
await swell.attributes.list({ limit: 100 });
```

**Response:**
```json
{
  "count": 1,
  "page_count": 1,
  "page": 1,
  "results": [
    {
      "visible": true,
      "values": [
        "Technivorm",
        "Aerobie",
        "Huskee",
        "Brewista",
        "Porlex"
      ],
      "type": "select",
      "searchable": false,
      "name": "Brand",
      "filterable": true,
      "content": {
        "feature": [
          "Algemeen"
        ]
      },
      "id": "brand"
    }
  ]
}
```

**Reply by logeshswell** [reply]
I don't have any ETA on showing up content editor at the moment. But, I will keep you posted as soon as it is available.
